src/main/java/uk/gov/hmcts/ccd/data/caseaccess/SwitchableCaseUserRepository.java

```diff
@@ -0,0 +1,85 @@
+package uk.gov.hmcts.ccd.data.caseaccess;
+
+import com.google.common.collect.Lists;
+import org.springframework.beans.factory.annotation.Qualifier;
+import uk.gov.hmcts.ccd.data.casedetails.CachedCaseDetailsRepository;
+import uk.gov.hmcts.ccd.data.casedetails.CaseDetailsRepository;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+import uk.gov.hmcts.ccd.domain.service.getcase.CaseNotFoundException;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+@Named
+@Singleton
+@Qualifier(SwitchableCaseUserRepository.QUALIFIER)
```

Comment:
Ok I can do that but it will diverge from the way we usually do it in code base and reduce consistency. How would you then want to differentiate inside the `SwitchableCaseUserRepository` bean as we need 2 different instances of `CaseUserRespository` beans? Don't we still need `@Qualifier` there?

Reply:
Or do you mean we remove the @Qualifier from the SwitchableCaseUserRepository and use @Primary there but use @Qualifier everywhere else? --> Yes.

Reply:
But feel free to ignore for now, due to relative priority of pending works.